From 72a6fcac0aa32b60f08948812ea905f20c7d9896 Mon Sep 17 00:00:00 2001
From: Sreedhar Govardhana <srgovard@amd.com>
Date: Tue, 20 Jul 2021 17:54:50 -0500
Subject: [PATCH] drivers/hwmon/pmbus: Adding support for XDPE192xxx and
 TDA38xxx VRs

Adding support for xdpe192xxx and tda38xxx voltage regulators to the
existing Infineon xdpe12284 driver.

Signed-off-by: Sreedhar Govardhana <srgovard@amd.com>
---
 drivers/hwmon/pmbus/xdpe12284.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/drivers/hwmon/pmbus/xdpe12284.c b/drivers/hwmon/pmbus/xdpe12284.c
index f8bc0f41cd5f..34589577f6e4 100644
--- a/drivers/hwmon/pmbus/xdpe12284.c
+++ b/drivers/hwmon/pmbus/xdpe12284.c
@@ -16,6 +16,8 @@
 #define XDPE122_PROT_VR12_5_10MV	0x02 /* VR12.5 mode, 10-mV DAC */
 #define XDPE122_PROT_IMVP9_10MV		0x03 /* IMVP9 mode, 10-mV DAC */
 #define XDPE122_AMD_625MV		0x10 /* AMD mode 6.25mV */
+#define XDPE19283A_VR			0x18 /* VOUT_MODE val for XDPE192xx */
+#define TDA38725_VR			0x0C /* VOUT_MODE val for TDA38xxx */
 #define XDPE122_PAGE_NUM		2

 static int xdpe122_read_word_data(struct i2c_client *client, int page,
@@ -99,6 +101,15 @@ static int xdpe122_identify(struct i2c_client *client,
 		case XDPE122_AMD_625MV:
 			info->vrm_version[i] = amd625mv;
 			break;
+		case XDPE19283A_VR:
+			info->format[PSC_VOLTAGE_OUT] = linear;
+			// <TBD> - May not be needed
+			info->vrm_version[i] = amd625mv;
+			break;
+		case TDA38725_VR:
+			// <TBD> - May not be needed
+			info->vrm_version[i] = amd625mv;
+			break;
 		default:
 			return -EINVAL;
 		}
@@ -142,6 +153,8 @@ static int xdpe122_probe(struct i2c_client *client)
 static const struct i2c_device_id xdpe122_id[] = {
 	{"xdpe12254", 0},
 	{"xdpe12284", 0},
+	{"xdpe19283", 0},
+	{"tds38725", 0},
 	{}
 };

@@ -150,6 +163,8 @@ MODULE_DEVICE_TABLE(i2c, xdpe122_id);
 static const struct of_device_id __maybe_unused xdpe122_of_match[] = {
 	{.compatible = "infineon,xdpe12254"},
 	{.compatible = "infineon,xdpe12284"},
+	{.compatible = "infineon,xdpe19283"},
+	{.compatible = "infineon,tds38725"},
 	{}
 };
 MODULE_DEVICE_TABLE(of, xdpe122_of_match);
--
2.25.1

